src/dependency-manager/config/component-context.ts

```diff
@@ -4,6 +4,16 @@ import { ServiceInterfaceConfig } from './service-config';
 
 export type OutputValue = string | number | boolean | null;
 
+export interface DatabaseContext {
+  protocol: string;
+  host: string;
+  port: number | string;
+  username: string;
+  password: string;
+  database: string;
+  connection_string: string;
```

Comment:
@mueschm and I discussed this at length, and connection string is the most widely accepted term for that value in the context of databases and ORMs so we decided to stick with it. It's better for it to align with expectations than to arbitrarily conform to a vague word like `url`.

Reply:
Can we support both? Seems pretty confusing to use a different value for interpolation compared to services.

Reply:
I disagree, but I'm never opposed to aliases like that. I'm happy to support both.

Reply:
Lets do both. Typeorm and Sequelize use `url` in the config and I find that more intuitive. Seems like that is because postgres/mysql npm pkgs also use url.

Reply:
So I added support for url via interpolation, but connection_string is the only valid option for the spec. I think it would be weird to have both there. Unless you guys disagree.

Reply:
That's fine. Host overrides for services should probably use the same mechanism, but we can cross that bridge later.